Show invalid combination of validation props

zcei opened the pull request:
As mentioned in #20, currently the validation message is not very meaningful when you have an additional un-allowed property in an otherwise valid validation object.

TimBeyer commented on lib/migration-payloads/validation/errors.js:
```diff
@@ -50,6 +50,10 @@ const errors = {
         return `A field can't have "${propName}" as a validation.`;
       },
+      INVALID_VALIDATION_PROPERTY_COMBINATION: (keys) => {
+        const formattedCombination = keys.map((key) => `"${key}"`).join(' & ');
```
I would try to find a way not to use the ampersand.

TimBeyer commented on lib/migration-payloads/validation/schema-validation.js:
```diff
@@ -132,6 +132,38 @@ const validateContentType = function (payload) {
+  return { message, path, type, context };
+};
+
+const combineErrors = function (fieldValidationsErrors) {
```
Maybe you could add a comment that shows the structure we're dealing with for every step of transformations so it's more easily understandable.

jfr3000 commented on lib/migration-payloads/validation/errors.js:
```diff
@@ -50,6 +50,10 @@ const errors = {
       },
+      INVALID_VALIDATION_PROPERTY_COMBINATION: (keys) => {
+        const formattedCombination = keys.map((key) => `"${key}"`).join(' & ');
+        return `A field can't have the combination ${formattedCombination} as a validation.`;
```
I'd phrase this more as "x is not a valid property for a validation" The combination thing is a bit confusing.

TimBeyer commented on test/unit/lib/migration-payloads/validation/payload-validation-field-validations.spec.js:
```diff
@@ -63,6 +63,32 @@ describe('payload validation', function () {
+        [
+          {
+            type: 'InvalidPayload',
+            message: `A field can't have the combination "size" & "somethingUnknown" as a validation.`
```
I find the error message somewhat misleading.  
I guess the problem is that our validations are just simple keys with the actual validation config being the value.  
So the error is more of a 'A validation can only have one key' error or something like that. 

I think ideally if there is one valid and `n` invalid properties we'd say that the valid validation cannot have any other keys in there, if all of them are invalid we just list them all as invalid, and if there are multiple valid ones we mention that they need to be separated.

At the same time it's all more of a nice to have, so I don't insist on it :)

zcei replied:
> A validation can only have one key

But a validation can have multiple keys, see #23.

TimBeyer replied:
Oh, right. Damn :(

zcei commented:
@TimBeyer addressed your feedback.

I think for now this should be sufficient - I'm playing with a different way of handling the additional properties, which would then allow us to access the list of allowed properties, so we could filter them and only display the *really* unknown ones.
